In [82]:
import pandas as pd
import numpy as np

In [83]:
df = pd.read_csv(r'C:\Users\harrycon\Documents\FoodBalanceSheets_E_Africa_NOFLAG.csv', encoding='latin-1')
df.head(10)

Area Code     Area  Item Code              Item  Element Code  \
0          4  Algeria       2501        Population           511   
1          4  Algeria       2501        Population          5301   
2          4  Algeria       2901       Grand Total           664   
3          4  Algeria       2901       Grand Total           674   
4          4  Algeria       2901       Grand Total           684   
5          4  Algeria       2903  Vegetal Products           664   
6          4  Algeria       2903  Vegetal Products           674   
7          4  Algeria       2903  Vegetal Products           684   
8          4  Algeria       2941   Animal Products           664   
9          4  Algeria       2941   Animal Products           674   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   
5           Food supply (kcal/capita/day)  kcal/capita/day   2932.00   
6  Protein supply quantity (g/capita/day)     g/capita/day     67.14   
7      Fat supply quantity (g/capita/day)     g/capita/day     53.84   
8           Food supply (kcal/capita/day)  kcal/capita/day    444.00   
9  Protein supply quantity (g/capita/day)     g/capita/day     27.76   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28  
5   2958.00   2941.00   2921.00   2932.00  
6     67.38     67.37     66.11     67.10  
7     54.72     52.04     55.59     54.57  
8    421.00    431.00    421.00    390.00  
9     26.97     27.35     26.71     24.73

In [84]:
df.shape


(60943, 12)

In [85]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 60943 entries, 0 to 60942
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Area Code     60943 non-null  int64  
 1   Area          60943 non-null  object 
 2   Item Code     60943 non-null  int64  
 3   Item          60943 non-null  object 
 4   Element Code  60943 non-null  int64  
 5   Element       60943 non-null  object 
 6   Unit          60943 non-null  object 
 7   Y2014         59354 non-null  float64
 8   Y2015         59395 non-null  float64
 9   Y2016         59408 non-null  float64
 10  Y2017         59437 non-null  float64
 11  Y2018         59507 non-null  float64
dtypes: float64(5), int64(3), object(4)
memory usage: 4.6+ MB


In [86]:
df.describe()

Area Code     Item Code  Element Code          Y2014          Y2015  \
count  60943.000000  60943.000000  60943.000000   59354.000000   59395.000000   
mean     134.265576   2687.176706   3814.856456     134.196282     135.235966   
std       72.605709    146.055739   2212.007033    1567.663696    1603.403984   
min        4.000000   2501.000000    511.000000   -1796.000000   -3161.000000   
25%       74.000000   2562.000000    684.000000       0.000000       0.000000   
50%      136.000000   2630.000000   5142.000000       0.090000       0.080000   
75%      195.000000   2775.000000   5511.000000       8.340000       8.460000   
max      276.000000   2961.000000   5911.000000  176405.000000  181137.000000   

               Y2016          Y2017          Y2018  
count   59408.000000   59437.000000   59507.000000  
mean      136.555222     140.917765     143.758381  
std      1640.007194    1671.862359    1710.782658  
min     -3225.000000   -1582.000000   -3396.000000  
25%         0.000000       0.000000       0.000000  
50%         0.080000       0.100000       0.070000  
75%         8.430000       9.000000       9.000000  
max    185960.000000  190873.000000  195875.000000

In [87]:
##QUESTION 1
#What is the total sum of Animal Fat produced in 2014 and 2017 respectively?

In [88]:

df2 = df.groupby('Item')['Y2014'].sum()
df2

Item
Alcohol, Non-Food        2403.00
Alcoholic Beverages    102410.11
Animal Products         11935.65
Animal fats            209460.54
Apples and products      9499.23
                         ...    
Vegetables, Other      155038.96
Vegetal Products       107145.19
Wheat and products     232670.13
Wine                     4497.36
Yams                   200396.96
Name: Y2014, Length: 119, dtype: float64


The total sum of  Aminal Fat produced in 2014 is 209460.54

In [89]:
df3 = df.groupby('Item')['Y2017'].sum()
df3

Item
Alcohol, Non-Food        2348.00
Alcoholic Beverages     95581.06
Animal Products         11547.65
Animal fats            269617.53
Apples and products     10198.90
                         ...    
Vegetables, Other      157752.59
Vegetal Products       107655.20
Wheat and products     240047.62
Wine                     4178.02
Yams                   229174.59
Name: Y2017, Length: 119, dtype: float64

The total sum of Aminal fat prouced in 2017 is 269617.53

In [ ]:
##Question 2

#What is the mean and standard deviation across the whole dataset for the year 2015 to 3 decimal places?

In [90]:

df.Y2015.mean()

135.23596649549236

In [91]:
df.Y2015.std()

1603.4039839423301

In [ ]:
##Question 3

#What is the total number and percentage of missing data in 2016 to 2 decimal places?

In [92]:

df.Y2016.isnull()

0        False
1        False
2        False
3        False
4        False
         ...  
60938    False
60939    False
60940    False
60941    False
60942    False
Name: Y2016, Length: 60943, dtype: bool

In [93]:
df.Y2016.isnull().sum()

1535

In [94]:
##Question 4

#What year has the highest sum of Import Quantity?

In [95]:

#Perform a groupby operation on ‘Element’ 

df4 = df.groupby('Element')['Y2014', 'Y2015', 'Y2016', 'Y2017', 'Y2018'].sum()
df4

<ipython-input-95-c412d19a83ac>:3: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  df4 = df.groupby('Element')['Y2014', 'Y2015', 'Y2016', 'Y2017', 'Y2018'].sum()


Y2014       Y2015       Y2016  \
Element                                                                      
Domestic supply quantity                1996716.35  2021493.55  2044842.70   
Export Quantity                          150020.64   157614.47   151920.46   
Fat supply quantity (g/capita/day)        10225.56    10235.74    10102.77   
Feed                                     216927.89   225050.22   228958.65   
Food                                    1212332.49  1232361.10  1247022.17   
Food supply (kcal/capita/day)            454257.00   453383.00   451810.00   
Food supply quantity (kg/capita/yr)       49650.63    49345.13    48985.28   
Import Quantity                          274144.48   267018.46   286582.78   
Losses                                   153223.00   155439.00   157787.00   
Other uses (non-food)                     78718.13    66254.41    69563.68   
Processing                               282923.00   287929.00   280631.00   
Production                              1931287.75  1947019.39  1943537.15   
Protein supply quantity (g/capita/day)    11836.46    11833.95    11779.69   
Residuals                                 30149.00    30045.00    37224.00   
Seed                                      21922.92    23976.82    23389.20   
Stock Variation                           58749.83    34910.99    33140.12   
Total Population - Both sexes           1031585.00  1058081.00  1085107.00   
Tourist consumption                         416.00      349.00       89.00   

                                             Y2017       Y2018  
Element                                                         
Domestic supply quantity                2088198.10  2161192.10  
Export Quantity                          182338.80   181594.80  
Fat supply quantity (g/capita/day)        10253.84    10258.69  
Feed                                     223705.68   233489.68  
Food                                    1258888.28  1303841.28  
Food supply (kcal/capita/day)            454681.00   455261.00  
Food supply quantity (kg/capita/yr)       48690.04    49056.85  
Import Quantity                          294559.09   287997.09  
Losses                                   160614.00   163902.00  
Other uses (non-food)                     91645.97    91300.97  
Processing                               292836.00   308429.00  
Production                              2030056.89  2075072.89  
Protein supply quantity (g/capita/day)    11842.45    11833.56  
Residuals                                 35500.00    34864.00  
Seed                                      24870.14    25263.14  
Stock Variation                           54316.91    20577.91  
Total Population - Both sexes           1112641.00  1140605.00  
Tourist consumption                          91.00       90.00

2017 has the highest sum for Importing Quantity.

In [96]:
##Question 5

##What is the total number of the sum of Production in 2014?

From the above we can see that total number of the sum of production in 2014 is 1931287.75

In [ ]:
##Quetion 6

#Which of these elements had the highest sum in 2018?


In [97]:

#Select columns ‘Y2018’ and ‘Element’, Perform a groupby operation on ‘Element’.

df_new = df[['Y2018', 'Element']]

df_new

Y2018                                 Element
0      42228.00           Total Population - Both sexes
1          0.00                Domestic supply quantity
2       3322.00           Food supply (kcal/capita/day)
3         91.83  Protein supply quantity (g/capita/day)
4         77.28      Fat supply quantity (g/capita/day)
...         ...                                     ...
60938     16.00                                    Food
60939      1.08     Food supply quantity (kg/capita/yr)
60940      1.00           Food supply (kcal/capita/day)
60941      0.04  Protein supply quantity (g/capita/day)
60942      0.01      Fat supply quantity (g/capita/day)

[60943 rows x 2 columns]

In [98]:
#perform a groupby operation on "Element"
df5 = df_new.groupby('Element')['Y2018'].sum()
df5

Element
Domestic supply quantity                  2161192.10
Export Quantity                            181594.80
Fat supply quantity (g/capita/day)          10258.69
Feed                                       233489.68
Food                                      1303841.28
Food supply (kcal/capita/day)              455261.00
Food supply quantity (kg/capita/yr)         49056.85
Import Quantity                            287997.09
Losses                                     163902.00
Other uses (non-food)                       91300.97
Processing                                 308429.00
Production                                2075072.89
Protein supply quantity (g/capita/day)      11833.56
Residuals                                   34864.00
Seed                                        25263.14
Stock Variation                             20577.91
Total Population - Both sexes             1140605.00
Tourist consumption                            90.00
Name: Y2018, dtype: float64

Domestic supply quantity had the highest sum in 2018 with  2161192.10


In [ ]:
##Question 7

#Which of these elements had the 3rd lowest sum in 2018?

#Food supply (kcal/capita/day)
#Food supply quantity (kg/capita/yr)
#Protein supply quantity (g/capita/day)
#Fat supply quantity (g/capita/day)
#Seed

From the above line we can see that fat supply quantity has the lowest sum in 2018 with 10258.69

In [ ]:
##Question 8

#What is the total Import Quantity in Algeria in 2018?

In [100]:
df6 = df.groupby(['Area', 'Element'] )['Y2018'].sum()
df6

Area      Element                               
Algeria   Domestic supply quantity                  90023.42
          Export Quantity                             811.54
          Fat supply quantity (g/capita/day)          309.09
          Feed                                      10668.00
          Food                                      63736.33
                                                      ...   
Zimbabwe  Protein supply quantity (g/capita/day)      176.57
          Residuals                                  -546.00
          Seed                                        140.00
          Stock Variation                            -447.00
          Total Population - Both sexes             14439.00
Name: Y2018, Length: 803, dtype: float64

In [101]:
Algeria = df6['Algeria']
Algeria

Element
Domestic supply quantity                  90023.42
Export Quantity                             811.54
Fat supply quantity (g/capita/day)          309.09
Feed                                      10668.00
Food                                      63736.33
Food supply (kcal/capita/day)             13288.00
Food supply quantity (kg/capita/yr)        1509.28
Import Quantity                           36238.29
Losses                                     7708.00
Other uses (non-food)                      3970.08
Processing                                 2479.00
Production                                60766.67
Protein supply quantity (g/capita/day)      367.28
Residuals                                   630.00
Seed                                        825.00
Stock Variation                            6172.00
Total Population - Both sexes             42228.00
Name: Y2018, dtype: float64

The total Import Quantity of Algeria in 2018 is 36238.29

In [102]:
#What is the total number of unique countries in the dataset?
df.Area.nunique()


49